src/Type/Doctrine/EntityManagerInterfaceThrowTypeExtension.php

```diff
@@ -0,0 +1,49 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Doctrine;
+
+use Doctrine\DBAL\Exception\UniqueConstraintViolationException;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Exception\ORMException;
+use Doctrine\Persistence\ObjectManager;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\DynamicMethodThrowTypeExtension;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use function array_map;
+
+class EntityManagerInterfaceThrowTypeExtension implements DynamicMethodThrowTypeExtension
+{
+
+	public const SUPPORTED_METHOD = [
+		'flush' => [
+			ORMException::class,
+			UniqueConstraintViolationException::class,
+		],
+	];
+
+	public function isMethodSupported(MethodReflection $methodReflection): bool
+	{
+		return $methodReflection->getDeclaringClass()->getName() === ObjectManager::class
+			&& isset(self::SUPPORTED_METHOD[$methodReflection->getName()]);
+	}
+
+	public function getThrowTypeFromMethodCall(MethodReflection $methodReflection, MethodCall $methodCall, Scope $scope): ?Type
+	{
+		$type = $scope->getType($methodCall->var);
+
+		if ((new ObjectType(EntityManagerInterface::class))->isSuperTypeOf($type)->yes()) {
+			return TypeCombinator::union(
+				...array_map(static function ($class): Type {
+					return new ObjectType($class);
+				}, self::SUPPORTED_METHOD[$methodReflection->getName()])
```

Comment:
This needs a test. You can create a new test case for the corresponding rule, and fetch the rule with `self::getContainer()->getByType(....`

Reply:
This seems covered by 
https://github.com/phpstan/phpstan-doctrine/blob/1.3.x/tests/Rules/Exceptions/data/unthrown-exception.php#L21-L25
to me.

I tried to comment the whole if in the `getThrowTypeFromMethodCall`, and tests were failing.

Reply:
If your `src/` does not change anything about test results then it's not needed.

Please show why your change is needed by writing a failing test first.

Reply:
I added another test, which fail without the change

Reply:
I added another test which fail without this change.